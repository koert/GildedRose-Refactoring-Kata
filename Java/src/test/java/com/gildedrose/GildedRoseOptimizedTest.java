package com.gildedrose;

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.Test;

class GildedRoseOptimizedTest {

    @Test
    void foo() {
        Item[] items = new Item[] {
            new Item("foo", -1, 0),
            new Item("foo", -1, 10),
            new Item("foo", 0, 10),
            new Item("foo", 10, 10)
        };
        GildedRoseOptimized app = new GildedRoseOptimized(items);
        app.updateQuality();
        assertEquals("foo", app.items[0].name);
        assertEquals(0, app.items[0].quality);
        assertEquals(8, app.items[1].quality);
        assertEquals(8, app.items[2].quality);
        assertEquals(9, app.items[3].quality);
        assertEquals(-2, app.items[0].sellIn);
        assertEquals(-2, app.items[1].sellIn);
        assertEquals(-1, app.items[2].sellIn);
        assertEquals(9, app.items[3].sellIn);
    }

    @Test
    void agedBrie() {
        Item[] items = new Item[] {
            new Item("Aged Brie", -1, 0),
            new Item("Aged Brie", 0, 0),
            new Item("Aged Brie", 10, 0),
            new Item("Aged Brie", 10, 50)
        };
        GildedRoseOptimized app = new GildedRoseOptimized(items);
        app.updateQuality();
        
        assertEquals("Aged Brie", app.items[0].name);
        assertEquals(2, app.items[0].quality);
        assertEquals(2, app.items[1].quality);
        assertEquals(1, app.items[2].quality);
        assertEquals(50, app.items[3].quality);
    }

    @Test
    void backstagePasses() {
        Item[] items = new Item[] {
            new Item("Backstage passes to a TAFKAL80ETC concert", 0, 10),
            new Item("Backstage passes to a TAFKAL80ETC concert", 5, 10),
            new Item("Backstage passes to a TAFKAL80ETC concert", 20, 10),
            new Item("Backstage passes to a TAFKAL80ETC concert", 10, 10)
        };
        GildedRoseOptimized app = new GildedRoseOptimized(items);
        app.updateQuality();
        
        assertEquals(0, app.items[0].quality);
        assertEquals(13, app.items[1].quality);
        assertEquals(11, app.items[2].quality);
        assertEquals(12, app.items[3].quality);
    }

    @Test
    void sulfuras() {
        Item[] items = new Item[] {
            new Item("Sulfuras, Hand of Ragnaros", 0, 10),
            new Item("Sulfuras, Hand of Ragnaros", 20, 10),
            new Item("Sulfuras, Hand of Ragnaros", 10, 10)
        };
        GildedRoseOptimized app = new GildedRoseOptimized(items);
        app.updateQuality();
        
        assertEquals(10, app.items[0].quality);
        assertEquals(10, app.items[1].quality);
        assertEquals(10, app.items[2].quality);
    }

    @Test
    void conjured() {
        Item[] items = new Item[] {
            new Item("Conjured", -1, 0),
            new Item("Conjured", -1, 10),
            new Item("Conjured", 0, 10),
            new Item("Conjured", 10, 10)
        };
        GildedRoseOptimized app = new GildedRoseOptimized(items);
        app.updateQuality();
        assertEquals("Conjured", app.items[0].name);
        assertEquals(0, app.items[0].quality);
        assertEquals(6, app.items[1].quality);
        assertEquals(6, app.items[2].quality);
        assertEquals(8, app.items[3].quality);
        assertEquals(-2, app.items[0].sellIn);
        assertEquals(-2, app.items[1].sellIn);
        assertEquals(-1, app.items[2].sellIn);
        assertEquals(9, app.items[3].sellIn);
    }

}
