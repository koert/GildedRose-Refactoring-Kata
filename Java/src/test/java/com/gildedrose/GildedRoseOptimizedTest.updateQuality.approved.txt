[foo, -1, 0] => foo, -2, 0 
[foo, -1, 10] => foo, -2, 8 
[foo, -1, 50] => foo, -2, 48 
[foo, 0, 0] => foo, -1, 0 
[foo, 0, 10] => foo, -1, 8 
[foo, 0, 50] => foo, -1, 48 
[foo, 4, 0] => foo, 3, 0 
[foo, 4, 10] => foo, 3, 9 
[foo, 4, 50] => foo, 3, 49 
[foo, 5, 0] => foo, 4, 0 
[foo, 5, 10] => foo, 4, 9 
[foo, 5, 50] => foo, 4, 49 
[foo, 6, 0] => foo, 5, 0 
[foo, 6, 10] => foo, 5, 9 
[foo, 6, 50] => foo, 5, 49 
[foo, 9, 0] => foo, 8, 0 
[foo, 9, 10] => foo, 8, 9 
[foo, 9, 50] => foo, 8, 49 
[foo, 10, 0] => foo, 9, 0 
[foo, 10, 10] => foo, 9, 9 
[foo, 10, 50] => foo, 9, 49 
[foo, 11, 0] => foo, 10, 0 
[foo, 11, 10] => foo, 10, 9 
[foo, 11, 50] => foo, 10, 49 
[Aged Brie, -1, 0] => Aged Brie, -2, 2 
[Aged Brie, -1, 10] => Aged Brie, -2, 12 
[Aged Brie, -1, 50] => Aged Brie, -2, 50 
[Aged Brie, 0, 0] => Aged Brie, -1, 2 
[Aged Brie, 0, 10] => Aged Brie, -1, 12 
[Aged Brie, 0, 50] => Aged Brie, -1, 50 
[Aged Brie, 4, 0] => Aged Brie, 3, 1 
[Aged Brie, 4, 10] => Aged Brie, 3, 11 
[Aged Brie, 4, 50] => Aged Brie, 3, 50 
[Aged Brie, 5, 0] => Aged Brie, 4, 1 
[Aged Brie, 5, 10] => Aged Brie, 4, 11 
[Aged Brie, 5, 50] => Aged Brie, 4, 50 
[Aged Brie, 6, 0] => Aged Brie, 5, 1 
[Aged Brie, 6, 10] => Aged Brie, 5, 11 
[Aged Brie, 6, 50] => Aged Brie, 5, 50 
[Aged Brie, 9, 0] => Aged Brie, 8, 1 
[Aged Brie, 9, 10] => Aged Brie, 8, 11 
[Aged Brie, 9, 50] => Aged Brie, 8, 50 
[Aged Brie, 10, 0] => Aged Brie, 9, 1 
[Aged Brie, 10, 10] => Aged Brie, 9, 11 
[Aged Brie, 10, 50] => Aged Brie, 9, 50 
[Aged Brie, 11, 0] => Aged Brie, 10, 1 
[Aged Brie, 11, 10] => Aged Brie, 10, 11 
[Aged Brie, 11, 50] => Aged Brie, 10, 50 
[Backstage passes to a TAFKAL80ETC concert, -1, 0] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 10] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 50] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 0] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 10] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 50] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 4, 0] => Backstage passes to a TAFKAL80ETC concert, 3, 3 
[Backstage passes to a TAFKAL80ETC concert, 4, 10] => Backstage passes to a TAFKAL80ETC concert, 3, 13 
[Backstage passes to a TAFKAL80ETC concert, 4, 50] => Backstage passes to a TAFKAL80ETC concert, 3, 50 
[Backstage passes to a TAFKAL80ETC concert, 5, 0] => Backstage passes to a TAFKAL80ETC concert, 4, 3 
[Backstage passes to a TAFKAL80ETC concert, 5, 10] => Backstage passes to a TAFKAL80ETC concert, 4, 13 
[Backstage passes to a TAFKAL80ETC concert, 5, 50] => Backstage passes to a TAFKAL80ETC concert, 4, 50 
[Backstage passes to a TAFKAL80ETC concert, 6, 0] => Backstage passes to a TAFKAL80ETC concert, 5, 2 
[Backstage passes to a TAFKAL80ETC concert, 6, 10] => Backstage passes to a TAFKAL80ETC concert, 5, 12 
[Backstage passes to a TAFKAL80ETC concert, 6, 50] => Backstage passes to a TAFKAL80ETC concert, 5, 50 
[Backstage passes to a TAFKAL80ETC concert, 9, 0] => Backstage passes to a TAFKAL80ETC concert, 8, 2 
[Backstage passes to a TAFKAL80ETC concert, 9, 10] => Backstage passes to a TAFKAL80ETC concert, 8, 12 
[Backstage passes to a TAFKAL80ETC concert, 9, 50] => Backstage passes to a TAFKAL80ETC concert, 8, 50 
[Backstage passes to a TAFKAL80ETC concert, 10, 0] => Backstage passes to a TAFKAL80ETC concert, 9, 2 
[Backstage passes to a TAFKAL80ETC concert, 10, 10] => Backstage passes to a TAFKAL80ETC concert, 9, 12 
[Backstage passes to a TAFKAL80ETC concert, 10, 50] => Backstage passes to a TAFKAL80ETC concert, 9, 50 
[Backstage passes to a TAFKAL80ETC concert, 11, 0] => Backstage passes to a TAFKAL80ETC concert, 10, 1 
[Backstage passes to a TAFKAL80ETC concert, 11, 10] => Backstage passes to a TAFKAL80ETC concert, 10, 11 
[Backstage passes to a TAFKAL80ETC concert, 11, 50] => Backstage passes to a TAFKAL80ETC concert, 10, 50 
[Sulfuras, Hand of Ragnaros, -1, 0] => Sulfuras, Hand of Ragnaros, -1, 0 
[Sulfuras, Hand of Ragnaros, -1, 10] => Sulfuras, Hand of Ragnaros, -1, 10 
[Sulfuras, Hand of Ragnaros, -1, 50] => Sulfuras, Hand of Ragnaros, -1, 50 
[Sulfuras, Hand of Ragnaros, 0, 0] => Sulfuras, Hand of Ragnaros, 0, 0 
[Sulfuras, Hand of Ragnaros, 0, 10] => Sulfuras, Hand of Ragnaros, 0, 10 
[Sulfuras, Hand of Ragnaros, 0, 50] => Sulfuras, Hand of Ragnaros, 0, 50 
[Sulfuras, Hand of Ragnaros, 4, 0] => Sulfuras, Hand of Ragnaros, 4, 0 
[Sulfuras, Hand of Ragnaros, 4, 10] => Sulfuras, Hand of Ragnaros, 4, 10 
[Sulfuras, Hand of Ragnaros, 4, 50] => Sulfuras, Hand of Ragnaros, 4, 50 
[Sulfuras, Hand of Ragnaros, 5, 0] => Sulfuras, Hand of Ragnaros, 5, 0 
[Sulfuras, Hand of Ragnaros, 5, 10] => Sulfuras, Hand of Ragnaros, 5, 10 
[Sulfuras, Hand of Ragnaros, 5, 50] => Sulfuras, Hand of Ragnaros, 5, 50 
[Sulfuras, Hand of Ragnaros, 6, 0] => Sulfuras, Hand of Ragnaros, 6, 0 
[Sulfuras, Hand of Ragnaros, 6, 10] => Sulfuras, Hand of Ragnaros, 6, 10 
[Sulfuras, Hand of Ragnaros, 6, 50] => Sulfuras, Hand of Ragnaros, 6, 50 
[Sulfuras, Hand of Ragnaros, 9, 0] => Sulfuras, Hand of Ragnaros, 9, 0 
[Sulfuras, Hand of Ragnaros, 9, 10] => Sulfuras, Hand of Ragnaros, 9, 10 
[Sulfuras, Hand of Ragnaros, 9, 50] => Sulfuras, Hand of Ragnaros, 9, 50 
[Sulfuras, Hand of Ragnaros, 10, 0] => Sulfuras, Hand of Ragnaros, 10, 0 
[Sulfuras, Hand of Ragnaros, 10, 10] => Sulfuras, Hand of Ragnaros, 10, 10 
[Sulfuras, Hand of Ragnaros, 10, 50] => Sulfuras, Hand of Ragnaros, 10, 50 
[Sulfuras, Hand of Ragnaros, 11, 0] => Sulfuras, Hand of Ragnaros, 11, 0 
[Sulfuras, Hand of Ragnaros, 11, 10] => Sulfuras, Hand of Ragnaros, 11, 10 
[Sulfuras, Hand of Ragnaros, 11, 50] => Sulfuras, Hand of Ragnaros, 11, 50 
[Conjured, -1, 0] => Conjured, -2, 0 
[Conjured, -1, 10] => Conjured, -2, 8 
[Conjured, -1, 50] => Conjured, -2, 48 
[Conjured, 0, 0] => Conjured, -1, 0 
[Conjured, 0, 10] => Conjured, -1, 8 
[Conjured, 0, 50] => Conjured, -1, 48 
[Conjured, 4, 0] => Conjured, 3, 0 
[Conjured, 4, 10] => Conjured, 3, 9 
[Conjured, 4, 50] => Conjured, 3, 49 
[Conjured, 5, 0] => Conjured, 4, 0 
[Conjured, 5, 10] => Conjured, 4, 9 
[Conjured, 5, 50] => Conjured, 4, 49 
[Conjured, 6, 0] => Conjured, 5, 0 
[Conjured, 6, 10] => Conjured, 5, 9 
[Conjured, 6, 50] => Conjured, 5, 49 
[Conjured, 9, 0] => Conjured, 8, 0 
[Conjured, 9, 10] => Conjured, 8, 9 
[Conjured, 9, 50] => Conjured, 8, 49 
[Conjured, 10, 0] => Conjured, 9, 0 
[Conjured, 10, 10] => Conjured, 9, 9 
[Conjured, 10, 50] => Conjured, 9, 49 
[Conjured, 11, 0] => Conjured, 10, 0 
[Conjured, 11, 10] => Conjured, 10, 9 
[Conjured, 11, 50] => Conjured, 10, 49 
